Source/SimpleActions/Private/SimpleAction.cpp

```diff
@@ -160,20 +207,35 @@ void USimpleAction::EndAction(bool bSuccess)
         return;
     }
 
+    // Kill all timers if the action is ending
+    if (StartDelay > 0.f) {
+        GetWorld()->GetTimerManager().ClearTimer(Timer_StartDelay);
+    }
+
+    if (DurationOverride > 0.f) {
+        GetWorld()->GetTimerManager().ClearTimer(Timer_DurationOverride);
+    }
+
     bActive = false;
 
     SetTickEnabled(false);
     UE_LOG(LogSimpleAction, Verbose, TEXT("EndAction: Stopping. Action '%s'. Success: %s."), *GetName(), bSuccess ? TEXT("True") : TEXT("False"));
 
-    ActionStop(bSuccess);
+    if(bStopWhenActiveStarted)
+    {
+		ActionStop(bSuccess);
+    }
+
+    if(bSuccess) { OnActionComplete.Broadcast(this); } else { OnActionFailed.Broadcast(this); }
```

Comment:
better not as one line

Reply:
I'm more of a fan of single line if it's a single purpose, looks neater when there are stacked statements. But true, there is only one and not stacked.